### PR TITLE
fix: omit `buildinfo_out` when `transpiler` and `declaration_transpiler` are used

### PR DESCRIPTION
**Pull request by walkerburgin**
Handle an edge case where a `.tsbuildinfo` file is expected but `tsc` is never used to emit anything.

Full repro here: https://github.com/walkerburgin/tsbuildinfo-repro

```
➜  tsbuildinfo-repro git:(master) ✗ bazel build //...
ERROR: /Users/wburgin/Repositories/walkerburgin/tsbuildinfo-repro/pkgs/bar-lib/BUILD:23:11: in ts_project rule //pkgs/bar-lib:bar-lib:
/Users/wburgin/Library/Caches/bazel/_bazel_wburgin/4ff382b5dbb2348818e8d30483df60bb/external/aspect_rules_ts+/ts/private/ts_project.bzl:70:5: The following files have no generating action:
pkgs/bar-lib/bar-lib.tsbuildinfo
ERROR: /Users/wburgin/Repositories/walkerburgin/tsbuildinfo-repro/pkgs/bar-lib/BUILD:23:11: Analysis of target '//pkgs/bar-lib:bar-lib' (config: ab1598d) failed
ERROR: Analysis of target '//pkgs/bar-lib:bar-lib_typecheck' failed; build aborted: Analysis failed
INFO: Elapsed time: 0.476s
INFO: 0 processes.
ERROR: Build did NOT complete successfully
```

We have `composite` projects for IDE support, use `swc` for transpilation, and are looking at using `tsgo` (or maybe OXC) for `.d.ts` transpilation.

---

### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: yes
- Breaking change (forces users to change their own code or config): no
- Suggested release notes appear below: yes
  
> Handle an edge case where a `.tsbuildinfo` file is expected but `tsc` is never used to emit anything.

### Test plan

- Added a new test case
- Manual testing; please provide instructions so we can reproduce:

See https://github.com/walkerburgin/tsbuildinfo-repro

### CODE REVIEW

**Comment by jbedard** [comment]
Would it be easy to add a test for this? I think a `ts_project` with the right attributes and a `build_test` would be enough?

File: examples/transpiler/BUILD.bazel [hunk 1]
```diff
@@ -349,3 +349,38 @@ assert_outputs(
         "transpiler/build-declarations_custom/a.js.map",
     ],
 )
+
+# custom js & dts transpilers (composite = true)
```
Comment (walkerburgin) [inline examples/transpiler/BUILD.bazel:353]: Before: 

Image

After: 

Image

**Comment by walkerburgin** [comment]
Sure, added

File: examples/transpiler/BUILD.bazel (hunk 1 above)
Comment (jbedard) [reply examples/transpiler/BUILD.bazel:353]: Perfect! 👍

**Review by jbedard — APPROVED** [review]

**Comment by walkerburgin** [comment]
@jbedard looks like this got stuck

**Comment by jbedard** [comment]
Thanks for the ping 👍
